lib/rx/util/health_check_authorization.rb

```diff
@@ -0,0 +1,35 @@
+module Rx
+  module Util
+    class HealthCheckAuthorization
+      def initialize(env, authorization)
+        @authorization = authorization
+        @env = env
+      end
+
+      def ok?
+        return custom_authorization(@env, @authorization) if @authorization.is_a?(Proc)
+
+        default_authorization(@authorization)
+      end
+
+      private
+
+      def custom_authorization(env, callable)
+        callable.call(env)
+      end
+
+      def default_authorization(key)
+        req_key = api_key(@env, "authorization_token")
+        req_key == key

```

Comment:
see my comment about using a `case` statement first, but if you decide not to go that direction I think this still reads easily without the assignment:

`key == api_key(@env, "authorization_token")`